src/lib/testing/utils.ts

```diff
@@ -1,17 +1,47 @@
 import { createDb, type DB } from '$lib/server';
 import { PostgreSqlContainer } from '@testcontainers/postgresql';
+import { Mutex } from 'async-mutex';
+import { sql } from 'drizzle-orm';
+import { test, type ExpectStatic } from 'vitest';
 
-export const createTestingDb = async <T>(generator?: (db: DB) => Promise<T>) => {
-    const container = new PostgreSqlContainer();
-    const startedContainer = await container.start();
-    const db = await createDb(startedContainer.getConnectionUri());
+type DbStateGenerator<T> = (db: DB) => Promise<T>;
+type TestFunc<T> = (args: {
+    db: DB;
+    generated?: Awaited<T>;
+    expect: ExpectStatic;
+}) => Promise<void>;
 
-    const generated = generator ? await generator(db) : null;
+const container = new PostgreSqlContainer();
+const startedContainer = await container.start();
+const db = await createDb(startedContainer.getConnectionUri());
+const lock = new Mutex();
 
-    return { container: startedContainer, db, generated };
-};
+const tableNames: string[] = [];
+for (const tableName in db._.tableNamesMap) {
+    if (tableName != '') tableNames.push(tableName);
+}
 
 export const mustGenerate = <T>(generated: T | null) => {
     if (!generated) process.exit(1);
```

Comment:
Can you make this throw rather than just end the program?

Reply:
Can indeed

Reply:
Excellent